src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,48 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class BasePage {
+    public WebDriver driver;
+    public WebDriverWait wait;
+
+    public BasePage(WebDriver driver) {
+        this.driver = driver;
+        wait = new WebDriverWait(driver, 10);
+    }
+
+    public void waitVisibility(By elementBy) {
+        wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(elementBy));
+    }
+
+    public void pageLoaded(){
+        wait.until(driver -> String
+                .valueOf(((JavascriptExecutor) driver).executeScript("return document.readyState"))
+                .equals("complete"));
+    }
+
+    public void click(By elementBy) {
+        waitVisibility(elementBy);
+        driver.findElement(elementBy).click();
+    }
+
+    public void writeText(By elementBy, String text) {
+        waitVisibility(elementBy);
+        driver.findElement(elementBy).sendKeys(text);
+    }
+
+    public String readText(By elementBy) {
+        waitVisibility(elementBy);
+        return driver.findElement(elementBy).getText();
+    }
+
+    public void enter(By elementBy) {
```

Comment:
Bad naming, because enter provides no context. pressEnterKey or something similar would be better.